alerts/index.js

```diff
@@ -0,0 +1,573 @@
+     */
+    const id = await alerts.getActiveAlertRefId(locationRefId);
+    if (!id) {
+        throw Error(`No active alert found for location ${locationRefId}!`);
+    }
+
+    await dbUtils.faunaQuery(
+        fq.Update(fq.Ref(fq.Collection("appointmentAlerts"), id), {
+            data: {
+                lastScraperRunRef: fq.Ref(
+                    fq.Collection("scraperRuns"),
+                    scraperRunRefId
+                ),
+            },
+        })
+    );
+    return id;
+}
+
+// TODO
+function maybeContinueAlerting() {
+    /* if appointments are still available, maybe send out more
+     * emails or texts?
+     */
+    return;
+}
+
+async function getLastAlertStartTime(locationRefId) {
+    /* find the entry in appointmentAlerts for this location with
+     * the lastScraperRunRef with the latest TS
+     */
+    const nsTimestamp = await dbUtils.faunaQuery(
+        fq.Select(
+            ["data", "startTime"],
+            fq.If(
+                fq.Exists(
+                    fq.Match(
+                        "appointmentAlertsByLocationRefSortByStartTime",
+                        fq.Ref(fq.Collection("locations"), locationRefId)
+                    )
+                ),
+                fq.Get(
+                    fq.Match(
+                        "appointmentAlertsByLocationRefSortByStartTime",
+                        fq.Ref(fq.Collection("locations"), locationRefId)
+                    )
+                ),
+                {
+                    // If we don't find a match, there has never been an alert for this location.
+                    // For ease of use, pretend there was one on Jan 1 2021.
+                    data: { startTime: fq.Time(moment("2021-01-01").format()) },
+                }
+            )
+        )
+    );
+    return moment(nsTimestamp.value);
+}
+
+async function setUpNewAlert(locationRefId, scraperRunRefId) {
+    // create a new entry in appointmentAlerts with locationRef and firstScraperRunRef set to scraperRunRef.
+    const record = await dbUtils.faunaQuery(
+        fq.Create(fq.Collection("appointmentAlerts"), {
+            data: {
+                locationRef: fq.Ref(fq.Collection("locations"), locationRefId),
+                firstScraperRunRef: fq.Ref(
+                    fq.Collection("scraperRuns"),
+                    scraperRunRefId
+                ),
+                startTime: fq.Now(),
+            },
+        })
+    );
+
+    return record.ref.id;
+}
+
+async function runImmediateAlerts(
+    locationRefId,
+    bookableAppointmentsFound,
+    availabilityWithNoNumbers,
+    parentIsChain
+) {
+    const location = await dbUtils
+        .retrieveItemByRefId("locations", locationRefId)
+        .then((res) => res.data);
+
+    let message;
+    if (bookableAppointmentsFound) {
+        message = `${bookableAppointmentsFound} appointments available at ${location.name} in ${location.address.city}. Visit macovidvaccines.com to book.`;
```

Comment:
- needs "https://" in our URL
- handle "1 appointments" singular/plural case.

Reply:
Just saw `SMALL_APPOINTMENT_NUMBER_THRESHOLD`, so singular/plural test not really needed

Reply:
> Just saw `SMALL_APPOINTMENT_NUMBER_THRESHOLD`, so singular/plural test not really needed

1 appointment caused just too many alerts... ugh.